### ISSUE TITLE
[ENHANCEMENT] Citation through Zotero API rather than BetterBibTex

### ISSUE THREAD

**Issue by pjalet** [Zettlr/Zettlr#1970]
## Description

I would like to cite references from my Zotero library directly (with the Zotero citation picker), without having to go through a file export by BetterBibTex.

## Proposed Changes

This change would improve several elements in the workflow:
- Speed and responsiveness: with the size of my main library (more than 8000 references), any update to the BetterBibTex export file is slow to generate and freezes Zettlr while reloading
- Easier synchronisation across computers: Zotero has a built-in synchronisation feature, whereas ensuring that the BetterBibTex export file is synchronised is difficult, which leads to broken references.
- The Zotero citation picker is flexible in the way it searches for references (you can combine authors' names with elements of the titles) whereas the BetterBibTex workflow is limited by what is included in the citation key.

Options to do this include:
- Using the BetterBibTeX web API (https://retorque.re/zotero-better-bibtex/citing/cayw/, used for example by https://github.com/mblode/vscode-zotero)
- Zotero's web API (https://www.zotero.org/support/dev/web_api/v3/start)
- Zotero's HTTP integration protocol (https://www.zotero.org/support/dev/client_coding/http_integration_protocol)

## Caveats

Using Zotero's web API will only work when users are online, which may be a limitation. Zotero-specific implementations will not benefit users of other citation managers, but there seems to be a large enough number of people using both Zettlr and Zotero to make the effort worthwhile.

**Comment by nathanlesage** [comment]
Good point. I don't think this will manage it into 2.0 because the citation provider is a fragile piece of code, but it's certainly a good improvement request!

**Comment by mahmoudjalloh** [comment]
Any progress on this? Currently having an update to my bibtex file through Zotero (7.0.32) freeze and sometimes crash Zettlr (4.1.1). Suggestions on workarounds also appreciated.

**Comment by nathanlesage** [comment]
Not yet. Implementing fetching citation keys from Zotero instead of a database is certainly possible. However, Pandoc will still require a database file, and so implementing this here would not really help.

**Comment by xell** [comment]
> Not yet. Implementing fetching citation keys from Zotero instead of a database is certainly possible. However, Pandoc will still require a database file, and so implementing this here would not really help.

Hi. In fact, Pandoc no longer needs a database file. Please see the zotero.lua solution [here](https://retorque.re/zotero-better-bibtex/exporting/pandoc/#from-markdown-to-zotero-live-citations). Simply put, download a tiny zotero.lua file, and use `pandoc --lua-filter=zotero.lua article.md -s -o article.docx` to generate the final docx document without any intermediate steps.

With this approach, I really hope Zettlr can leverage Zotero’s native citation picker. Thanks.
